web_tree_dynamic_colored_field/static/src/js/web_tree_dynamic_colored_field.js

```diff
@@ -0,0 +1,119 @@
+odoo.define('web_tree_dynamic_colored_field', function (require) {
+    'use strict';
+
+    var ListRenderer = require('web.ListRenderer');
+    var pyeval = require('web.pyeval');
+
+    ListRenderer.include({
+        /**
+         * @override
+         */
+        _renderBody: function () {
+            this.colorField = this.arch.attrs.colors.split(';')
+                .filter(color => color.startsWith('color_field'))[0]
+                .split(':')[1]
+                .trim();
+            if (this.colorField) {
+                // validate the presence of that field in tree view
+                var fieldNames = _(this.columns).map(
+                    (value) => { return value.attrs.name; }
+                );
+                if (fieldNames.indexOf(this.colorField) === -1) {
```

Comment:
I'd say you should just log here. If you keep it you should use `_t` to translate the msg.